rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+                            current_ans_lookups,
+                            conn,
+                        )
+                    },
+                    WriteSetChange::DeleteTableItem(delete_table_item) => {
+                        let table_handle = delete_table_item.handle.as_str();
+                        if table_handle != ans_name_records_table_handle.as_str() {
+                            continue;
+                        }
+                        Self::from_name_record_table_delete_table_item_v1(
+                            delete_table_item,
+                            txn_version,
+                            txn_timestamp,
+                        )
+                    },
+                    _ => continue,
+                };
+                match res {
+                    Ok((current_ans_lookups, ans_lookups)) => {
+                        new_current_ans_lookups.extend(current_ans_lookups);
+                        new_ans_lookups.extend(ans_lookups);
+                    },
+                    Err(e) => {
+                        tracing::error!("Failed to parse name record table item. Error: {:?}", e);
+                        continue;
+                    },
+                }
+            }
+
+            // Extract primary name reverse lookup changes
+            for wsc in &transaction_info.changes {
+                let res = match wsc.change.as_ref().unwrap() {
+                    WriteSetChange::WriteTableItem(write_table_item) => {
+                        let table_handle = write_table_item.handle.as_str();
+                        if table_handle != ans_primary_names_table_handle.as_str() {
+                            continue;
+                        }
+                        Self::from_primary_name_record_table_write_table_item_v1(
+                            write_table_item,
+                            txn_version,
+                            txn_timestamp,
+                            address_to_primary_name_record,
+                            conn,
+                        )
+                    },
+                    WriteSetChange::DeleteTableItem(delete_table_item) => {
+                        let table_handle = delete_table_item.handle.as_str();
+                        if table_handle != ans_primary_names_table_handle.as_str() {
+                            continue;
+                        }
+                        Self::from_primary_name_record_table_delete_table_item_v1(
+                            delete_table_item,
+                            txn_version,
+                            txn_timestamp,
+                            address_to_primary_name_record,
+                            conn,
+                        )
+                    },
+                    _ => continue,
+                };
+
+                match res {
+                    Ok((current_ans_lookups, ans_lookups)) => {
+                        for (record_key, name_record) in current_ans_lookups {
```

Comment:
This flow is hard to understand. I think that you may be able to do this differently if you first do a full loop to get the full mapping similar to how we deal with object parsing and then do a full loop to get both NR and PN. I roughly understand what you're doing here but it's very confusing.